packages/vaadin-avatar/test/avatar.test.js

```diff
@@ -189,6 +192,39 @@ describe('vaadin-avatar', () => {
         expect(avatar.getAttribute('title')).to.equal('anonymous');
       });
     });
+
+    describe('img fallback', () => {
+      beforeEach(() => {
+        sinon.stub(console, 'warn');
+      });
+      afterEach(() => {
+        console.warn.restore();
+      });
+
+      it('should display abbr as fallback if image can not be loaded', async () => {
+        avatar.abbr = 'YY';
+        avatar.img = invalidImageSrc;
+        await oneEvent(imgElement, 'error');
+        expect(imgElement.hasAttribute('hidden')).to.be.true;
+        expect(abbrElement.hasAttribute('hidden')).to.be.false;
+      });
+
+      it('should log a warning if image can not be loaded', async () => {
+        avatar.img = invalidImageSrc;
+        await oneEvent(imgElement, 'error');
+        expect(console.warn.calledOnce).to.be.true;
+      });
+
+      it('should display img when setting a valid source after setting an invalid source', async () => {
+        avatar.abbr = 'YY';
+        avatar.img = invalidImageSrc;
+        await oneEvent(imgElement, 'error');
+        avatar.img = validImageSrc;
+        await oneEvent(imgElement, 'load');
+        expect(imgElement.hasAttribute('hidden')).to.be.false;
+        expect(abbrElement.hasAttribute('hidden')).to.be.true;
+      });
```

Comment:
We can also add a test to verify that abbreviation from the name is used when image cannot be loaded and `abbr` is not provided.

Reply:
Sure, added tests for when a name is provided and when no other data is provided (=icon visible).